### PR TITLE
CA-337113: Turn create_tools_sr into remove_tools_sr

### PR DESCRIPTION
**Pull request by robhoes**
The Tools ISO has been removed from dom0, so there is no need anymore for the
Tools SR.

The removal of the Tools SR is best-effort: a warning is logged if it fails (as
done before for redundant SRs). Failure occurs if there is still some VM that
has a Tools VDI attached to it. The cleanup code runs each time the master xapi
start, so once this situation has been resolved, the SR will eventually get
cleaned up.

Signed-off-by: Rob Hoes <rob.hoes@citrix.com>

### CODE REVIEW

File: ocaml/tests/suite_alcotest.ml [hunk 1]
```diff
@@ -71,7 +71,6 @@ let () =
       @ Test_vm.tests
       @ Test_cpuid_helpers.tests
       @ Test_pool_cpuinfo.tests
-      @ Test_dbsync_master.tests
```
Comment (lippirk) [inline ocaml/tests/suite_alcotest.ml:74]: this made me suspicious, but it looks like the dbsync master tests were actually only testing the tools sr :heavy_check_mark:

File: ocaml/xapi/dbsync_master.ml [hunk 2]
```diff
@@ -107,90 +107,23 @@ let release_locks ~__context =
...
-    )
+  let srs = Db.SR.get_all ~__context in
+  let tools_srs =
+    List.filter (fun self -> Db.SR.get_is_tools_sr ~__context ~self) srs in
+  let old_srs =
+    List.filter (fun self ->
+        let other_config = Db.SR.get_other_config ~__context ~self in
+        (Db.SR.get_is_tools_sr ~__context ~self = false) &&
+        ((List.mem_assoc Xapi_globs.tools_sr_tag other_config) ||
+         (List.mem_assoc Xapi_globs.xensource_internal other_config))
```
Comment (lippirk) [inline ocaml/xapi/dbsync_master.ml:123]: why does having a `xensource_internal` property mean it's an old sr?

Comment (robhoes) [reply ocaml/xapi/dbsync_master.ml:123]: It's from the days before we had a proper `is_tools_sr` field, and even from before the days that we had a `tools_sr_tag` other_config key.

Comment (psafont) [reply ocaml/xapi/dbsync_master.ml:123]: this check has just been moved, I'd rather keep the logic the same in here unless there's a reason to change it.

Comment (robhoes) [reply ocaml/xapi/dbsync_master.ml:123]: Indeed.

**Pull request merged**